pkg/controller/handlers/mcpcatalog/mcpcatalog.go

```diff
@@ -248,142 +246,30 @@ func (h *Handler) readMCPCatalogDirectory(catalog string) ([]types.MCPServerCata
-		"spec.mcpCatalogName": catalog.Name,
-	}); err != nil {
-		return fmt.Errorf("failed to list entries: %w", err)
-	}
-
-	// TODO(g-linville): if this is too inefficient, we can do it in a handler for individual MCPServerCatalogEntry objects instead.
-	// Then we would only need to loop over servers, and not over entries also.
-	for _, entry := range entries.Items {
-		var servers v1.MCPServerList
-		if err := req.Client.List(req.Ctx, &servers, client.InNamespace(req.Namespace), client.MatchingFields{
-			"spec.mcpServerCatalogEntryName": entry.Name,
-		}); err != nil {
-			return fmt.Errorf("failed to list servers: %w", err)
-		}
-
-		for _, server := range servers.Items {
-			// Admin users can run whatever they want, so don't shut down any of their servers.
-			if user, err := h.gatewayClient.UserByID(req.Ctx, server.Spec.UserID); err == nil && user.Role == types.RoleAdmin {
-				continue
-			}
-
-			if _, ok := allowedUserIDs[server.Spec.UserID]; !ok {
-				if err := req.Client.Delete(req.Ctx, &server); err != nil {
-					return fmt.Errorf("failed to delete server %s: %w", server.Name, err)
-				}
-			}
-		}
-	}
-
-	return nil
-}
-
 func (h *Handler) SetUpDefaultMCPCatalog(ctx context.Context, c client.Client) error {
-	if h.defaultCatalogPath == "" {
-		// Delete it if it exists.
-		var catalog v1.MCPCatalog
-		if err := c.Get(ctx, router.Key(system.DefaultNamespace, "default"), &catalog); err == nil {
-			if err := c.Delete(ctx, &catalog); err != nil {
-				return fmt.Errorf("failed to delete default catalog: %w", err)
-			}
-		}
+	var existing v1.MCPCatalog
+	if err := c.Get(ctx, router.Key(system.DefaultNamespace, system.DefaultCatalog), &existing); err == nil {
+		// Default catalog already exists, do nothing.
 		return nil
 	}
 
-	var existing v1.MCPCatalog
-	if err := c.Get(ctx, router.Key(system.DefaultNamespace, "default"), &existing); err == nil {
-		// See if the URL has changed.
-		if len(existing.Spec.SourceURLs) > 0 && existing.Spec.SourceURLs[0] != h.defaultCatalogPath {
-			existing.Spec.SourceURLs = []string{h.defaultCatalogPath}
-			if err := c.Update(ctx, &existing); err != nil {
-				return fmt.Errorf("failed to update default catalog: %w", err)
-			}
-		}
-		return nil
+	sourceURLs := []string{}
+	if h.defaultCatalogPath != "" {
+		sourceURLs = []string{h.defaultCatalogPath}
 	}
 
 	if err := c.Create(ctx, &v1.MCPCatalog{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "default",
+			Name:      system.DefaultCatalog,
 			Namespace: system.DefaultNamespace,
 		},
 		Spec: v1.MCPCatalogSpec{
-			DisplayName:    "Default",
-			SourceURLs:     []string{h.defaultCatalogPath},
-			AllowedUserIDs: []string{"*"},
-			IsReadOnly:     true,
+			DisplayName: "Default",
+			SourceURLs:  sourceURLs,
 		},
 	}); err != nil {
 		return fmt.Errorf("failed to create default catalog: %w", err)
 	}
 
 	return nil
 }
-
-func (h *Handler) SetUpUserCatalogAuthorizations(req router.Request, _ router.Response) error {
```

Comment:
`UserCatalogAuthorizations` were an old type that we used to check authz logic that is no longer needed.